persistence/sql/persister_identity.go

```diff
@@ -472,3 +510,115 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
+					q = q.Where(stmt, value)
+				}
+			}
+			return q
+		case "postgres", "cockroach":
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`%s @> '"%s"'`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`%s @> '{"%s":"%s"}'`, field, innerField, value))
+				}
+			}
+			return q
+		}
+		return q
+	}
+}
+
+func (p *Persister) searchCredentialQuery(field string, values []string) pop.ScopeFunc {
+	var innerField string
+	_, innerField = extractFieldAndInnerFields(field)
+	switch innerField {
+	case "type":
+		return func(q *pop.Query) *pop.Query { return q.Where("credential_types.name IN (?)", values) }
+	case "identifier":
+		return func(q *pop.Query) *pop.Query { return q.Where("credential_identifiers.identifier IN (?)", values) }
+	default:
+		return func(q *pop.Query) *pop.Query { return q }
+	}
+}
+
+var quoteChar = map[string]uint8{
+	"cockroach": '"',
+	"mariadb":   '`',
+	"mysql":     '`',
+	"postgres":  '"',
+	"sqlite3":   '"',
+}
+
+func (p *Persister) Quote(key string, c uint8) string {
+	if c == 0 {
+		c = '"'
+	}
+	parts := strings.Split(key, ".")
+
+	for i, part := range parts {
+		part = strings.Trim(part, `"`)
+		part = strings.Trim(part, "`")
+		part = strings.TrimSpace(part)
+
+		parts[i] = fmt.Sprintf(`%c%v%c`, c, part, c)
+	}
+
+	return strings.Join(parts, ".")
+}
+func (p *Persister) buildScope(queryValues url.Values) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		for field, values := range queryValues {
+			if IsStringInSlice([]string{"page", "per_page"}, field) {
+				continue
+			}
+			if IsStringInSlice([]string{"with_credentials"}, field) {
+				q = q.LeftJoin("identity_credentials ic", "identities.id=ic.identity_id")
+				q = q.LeftJoin("identity_credential_types credential_types", "credential_types.id=ic.identity_credential_type_id")
+				q = q.LeftJoin("identity_credential_identifiers credential_identifiers", "credential_identifiers.identity_credential_id=ic.id")
+				//q = q.LeftJoin("identity_credentials credentials", "credentials.identity_id=identities.id")
+				continue
+			}
+			if strings.HasPrefix(field, "traits") {
+				q = q.Scope(p.getJsonSearchQuery(field, values))
+				continue
+			}
+			if strings.HasPrefix(field, "credentials") {
+				q = q.Scope(p.searchCredentialQuery(field, values))
+				continue
+			}
+			//q = q.Where("? = ?", field, values[0])
+			field = p.Quote(field, quoteChar[p.Connection(context.Background()).Dialect.Name()])
+			q = q.Where(fmt.Sprintf("%s IN (?)", field), values)
+		}
+		return q
+	}
+}
+
+func extractFieldAndInnerFields(field string) (string, string) {
+	if !strings.Contains(field, ".") {
+		return field, ""
+	}
+	dotIndex := strings.Index(field, ".")
+	return field[:dotIndex], field[dotIndex+1:]
+}
+
+func IsStringInSlice(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
https://github.com/ory/x/blob/a065f5780b6040793599f5d9fd68a246322b56ab/stringslice/has.go :wink:

Reply:
ha ha ha ok I switch this